docs/4.2/ssh_azuread.md

```diff
@@ -0,0 +1,190 @@
+   
+9. Edit User Attributes & Claims
+
+a. Edit the Claim Name Change the name identifier format to Default Make sure the source attribute is user.userprincipalname
+   ![Confirm Name Identifier](img/azuread/azuread-9a-nameidentifier.png)
+   
+b. Add a group Claim to have user security groups groups
+   ![Put in Security group claim](img/azuread/azuread-9b-groupclaim.png)
+   
+10. On the SAM Signing Certificate selec to SAML Download the Federation Metadata XML.  
+   ![Download Federation Metadata XML](img/azuread/azuread-10-fedmeatadataxml.png)
+
+!!! warning "Important"  This a imporant document.  Treat this file as you would a password.
+
+## Create a SAML Connector
+
+Now, create a SAML connector [resource](admin-guide.md#resources).  Replace the <proxy address> with your teleport address and the group ids with the AzureAD group values.
+Write down this template as `azure-connector.yaml`:
+
+```yaml
+kind: saml
+version: v2
+metadata:
+  # the name of the connector
+  name: saml
+spec:
+  display: "Microsoft"
+  # acs is the Assertion Consumer Service URL. This should be the address of
+  # the Teleport proxy that your identity provider will communicate with.
+  acs: https://<proxy address>:3080/v1/webapi/saml/acs
+  attributes_to_roles:
+    - {name: "http://schemas.microsoft.com/ws/2008/06/identity/claims/groups", value: "<group id 930210...>", roles: ["admin"]}
+    - {name: "http://schemas.microsoft.com/ws/2008/06/identity/claims/groups", value: "<group id 93b110...>", roles: ["dev"]}
+  entity_descriptor: |
+    <federationmedata.xml contents>
+```
+
+Create the connector using `tctl` tool:
+
+```bsh
+$ tctl create azure-connector.yaml
+```
+
+## Create Teleport Roles
+
+We are going to create 2 roles:
+
+-  Privileged role admin who is able to login as root and is capable of administrating
+the cluster
+- Non-privileged dev
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: admin
+spec:
+  options:
+    max_session_ttl: 24h
+  allow:
+    logins: [root]
+    node_labels:
+      "*": "*"
+    rules:
+      - resources: ["*"]
+        verbs: ["*"]
+```
+
+Devs are only allowed to login to nodes labeled with `access: relaxed`
+Teleport label. Developers can log in as either `ubuntu` or a username that
+arrives in their assertions. Developers also do not have any rules needed to
+obtain admin access to Teleport.
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: dev
+spec:
+  options:
+    max_session_ttl: 24h
+  allow:
+    logins: [ "{% raw %}{{external.username}}{% endraw %}", ubuntu ]
+    node_labels:
+      access: relaxed
+```
+
+**Notice:** Replace `ubuntu` with linux login available on your servers!
+
+```bsh
+$ tctl create admin.yaml
+$ tctl create dev.yaml
+```
+
+## Testing
+![Login with Microsoft](img/azuread/azure-11-loginwithmsft.png)
+
+
+The Web UI will now contain a new button: "Login with Microsoft". The CLI is
+the same as before:
+
+```bsh
+$ tsh --proxy=proxy.example.com login
```

Comment:
This only works if you change your Teleport config:

```yaml
auth_service:
  authentication:
    type: saml
```

Otherwise you need to use`tsh --proxy=proxy.example.com --auth=azure-saml login`

Reply:
provided instructions on changing the default authentication type